src/analysis/ConfigurationsExt.cpp

```diff
@@ -304,6 +306,209 @@ namespace espressopp {
+                  stat = req.wait();
+                  nIds = *stat.count<int>();
+
+                  req = system.comm->irecv<size_t>(iproc, DEFAULT_TAG, types, maxN);
+                  system.comm->send(iproc, DEFAULT_TAG, 0);
+                  stat = req.wait();
+                  nTypes = *stat.count<size_t>();
+
+                  req = system.comm->irecv<real>(iproc, DEFAULT_TAG, coordinates, 3*maxN);
+                  system.comm->send(iproc, DEFAULT_TAG, 0);
+                  stat = req.wait();
+                  nCoords = *stat.count<real>();
+
+                  // make sure to have 3 coordinate values for each id
+
+                  if (nCoords != 3 * nIds) {
+                    LOG4ESPP_ERROR(logger, "serious error collecting data, got " <<
+                                  nIds << " ids, but " << nCoords << " coordinates");
+                  }
+
+
+                  req = system.comm->irecv<real>(iproc, DEFAULT_TAG, velocities, 3*maxN);
+                  system.comm->send(iproc, DEFAULT_TAG, 0);
+                  stat = req.wait();
+                  nVelocs = *stat.count<real>();
+
+                  // make sure to have 3 velocities values for each id
+
+                  if (nVelocs != 3 * nIds) {
+                    LOG4ESPP_ERROR(logger, "serious error collecting data, got " <<
+                                  nIds << " ids, but " << nVelocs << " velocities");
+                  }
+
+
+                  nother = nIds;
+
+               } else {
+                 nother = myN;
+               }
+
+
+               LOG4ESPP_INFO(logger, "add " << nother << " coordinates of proc " << iproc);
+
+               for (int i = 0; i < nother; i++) {
+
+                 int index = ids[i];
+                 size_t tipo = types[i];
+
+                 LOG4ESPP_INFO(logger, "set coordianates of particle with id " << index << " and type " << tipo <<
+                                       ": " << coordinates[3*i] << " " <<  coordinates[3*i+1] << " " << coordinates[3*i+2] <<
+                                       "and velocities: " <<  velocities[3*i] << " " <<  velocities[3*i+1] << " " << velocities[3*i+2]);
+
+
+                 RealND _vec(6);
+                 for (int k=0; k<3; k++)
+                   _vec.setItem(k, coordinates[3*i + k]);
+
+                 for (int k=0; k<3; k++)
+                   _vec.setItem(k+3, velocities[3*i + k]);
+                 //config->set(index, _vec);
+                 config->set_id_vec_type(index, _vec, tipo);
+               }
+            }
+
+            LOG4ESPP_INFO(logger, "save the latest configuration");
+
+            pushConfig(config);
+
+          } else {
+
+           LOG4ESPP_INFO(logger, "proc " << system.comm->rank() << " sends data "
+                          << " of " << myN << " particles");
+
+           // not master process, send data to master process
+
+           int tmp;
+
+           boost::mpi::status stat;
```

Comment:
Currently not used. Can be added along the lines of similar code above in the same method.